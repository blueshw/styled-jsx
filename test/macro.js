// Packages
import test from 'ava'

// Ours
import macros from 'babel-plugin-macros'
import jsx from 'babel-plugin-syntax-jsx'
import _transform from './_transform'

const transform = (file, opts = {}) =>
  _transform(file, {
    plugins: [macros, jsx],
    ...opts
  })

test('transpiles correctly', async t => {
  const { code } = await transform('./fixtures/macro.js')
  t.snapshot(code)
})
