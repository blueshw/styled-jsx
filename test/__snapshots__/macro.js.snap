// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transpiles correctly 1`] = `
"

const { className, styles } = {
  styles: <_JSXStyle styleId={\\"2052294191\\"} css={\\"div.jsx-2052294191{color:red;}\\"} />,
  className: 'jsx-2052294191'
};

const dynamicStyles = props => ({
  styles: <_JSXStyle styleId={\\"290194820\\"} css={\`div.__jsx-style-dynamic-selector{color:\${props.color};}\`} dynamic={[props.color]} />,
  className: _JSXStyle.dynamic([['290194820', [props.color]]])
});

const test = {
  styles: <_JSXStyle styleId={\\"2052294191\\"} css={\\"div.jsx-2052294191{color:red;}\\"} />,
  className: 'jsx-2052294191'
};

const dynamicStyles2 = props => ({
  styles: <_JSXStyle styleId={\\"290194820\\"} css={\`div.__jsx-style-dynamic-selector{color:\${props.color};}\`} dynamic={[props.color]} />,
  className: _JSXStyle.dynamic([['290194820', [props.color]]])
});

const ExampleComponent = props => {
  const { className, styles } = dynamicStyles(props);

  return <div className={className}>
      howdy
      {styles}
    </div>;
};"
`;
